data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/repository/OrgRepository.java

```diff
@@ -0,0 +1,17 @@
+package gov.cdc.etldatapipeline.organization.repository;
+
+import gov.cdc.etldatapipeline.organization.model.dto.org.OrganizationSp;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.Set;
+
+@Repository
+public interface OrgRepository extends JpaRepository<OrganizationSp, String> {
+
+    @Query(nativeQuery = true, value = "execute sp_organization_event :org_uids")
+    Set<OrganizationSp> computeAllOrganizations(@Param("org_uids") String org_ids);
```

Comment:
Is there a reason to use `Set` here? general practice is to use `Optional` for the output returned by repo method.

Reply:
Optional is usually for the non-collection instances to anticipate null objects. The stored proc returns a collection of objects and `set`/`list` can be used.
https://stackoverflow.com/questions/53085781/java-8-optional-class-on-list-of-objects